frontend/src/dashboard.js

```diff
@@ -116,23 +133,59 @@ export class Dashboard extends React.Component {
     HttpClient.get([Settings.serverUrl, 'dashboard'], params)
       .then(response => HttpClient.handleResponse(response))
       .then(data => {
-        this.setState({dashboards: data['dashboards'], filteredDashboards: data['dashboards']}, this.getWidgets);
+        this.setState({dashboards: data['dashboards'], filteredDashboards: data['dashboards']});
       });
   }
 
-  getWidgets() {
+  getDefaultDashboard = (handledObject = null) => {
+    const { primaryObject, activeDashboard, setActiveDashboard } = this.context;
+    const paramProject = this.props.params?.project_id;
+
+    let targetObject = handledObject ?? primaryObject ?? paramProject;
+
+    if (typeof(targetObject) === 'string') {
+      HttpClient.get([Settings.serverUrl, 'project', paramProject])
+        .then(response => HttpClient.handleResponse(response))
+        .then(data => {
+          targetObject = data;
+        });
+
+    }
+
+    if ( !activeDashboard && targetObject?.defaultDashboard ){
+      setActiveDashboard(targetObject.defaultDashboard);
+      this.setState({
+        'selectedDashboard': targetObject.defaultDashboard,
+        'dashboardInputValue': targetObject.defaultDashboard?.title
+      })
+    } else {
+      this.setState({
+        'selectedDashboard': 'Select a dashboard',
+        'dashboardInputValue': 'Select a dashboard'
```

Comment:
See my comment about placeholder (comment on same file). there should be no Input default value as it looks a bit odd. Instead of this we use placeholder. 
Right now it looks a bit odd as something is selected there but I have nothing:
![image](https://github.com/user-attachments/assets/e5ce8640-751d-40bf-96ac-382cdad929d8)